### ISSUE TITLE
provider/vcd: When creating a vcd_vapp resource using DHCP we should wait for the IP Address

### ISSUE THREAD

**Issue by robcoward** [hashicorp/terraform#5159]
If the ip property is not passed to a vcd_vapp resource, the resulting VM will use DHCP to allocate an IP Address. Initially the vCloud Director UI will not show an IP Address for the new VM, however once the VM has booted and the open-vm-tools package started, it updates vCloud Director with the DHCP allocated IP Address. 

If you subsequently run a "terraform refresh", the resulting tfstate file does also include the correct address in the ip property for the vapp.

It would be really handy if when creating the vcd_vapp resource, if no ip address is specified then it should wait until the vm's open-vm-tools has updated vCloud Director with it's allocated IP address and pull it back. It would open up all sorts of possibilities, not the least being able to configure connection details to bounce off bastion hosts, something like:
```
resource "vcd_vapp" "webservers" {
    name          = "${format("web%02d", count.index + 1)}"
    count         = "${var.webserver_count}"
    ....
    depends_on    = [ "vcd_vapp.jumpbox" ]
    connection {
        bastion_host = "${var.jumpbox_ext_ip}"
        host         = "${vcd_vapp.webservers.ip}"
    }
    provisioner "chef"  {
        ....
    }            
}
```

**Comment by robcoward** [comment]
Hi @devopsbrett how much effort would it be to poll vCloud Director and wait for a VM's ip address to be populated ?

**Comment by devopsbrett** [comment]
Hi @robcoward I'm going to look at this now

**Comment by jen20** [comment]
Fixed in #5195.

**Comment by robcoward** [comment]
@devopsbrett thanks for the quick turn around.
